### PR TITLE
Add maxWarnings CLI option

### PR DESCRIPTION
**Pull request by house9**
Exit with non-zero value when the number of warnings is greater than the maxWarning option.

see #567 

DCO 1.1 Signed-off-by: Jesse House <jesse.house@codingzeal.com>

### CODE REVIEW

**Comment by DanPurdy** [comment]
Thanks, I'll have a look over this later :+1: 

closes #567

**Comment by DanPurdy** [comment]
Hi @house9 Sorry for the delay here, we just finally merged all of our 1.6 work into develop which will cause a few conflicts for this PR. Would you mind updating it now and then we can look to get it merged in.

Thanks! :+1:

File: bin/sass-lint.js [hunk 1]
```diff
@@ -10,6 +10,12 @@ var configPath,
     configOptions = {},
     exitCode = 0;
 
+var tooManyWarnings = function (detects) {
+  var warningCount = lint.warningCount(detects).count;
+
+  return warningCount > 0 && warningCount > program.maxWarnings;
+};
+
 var detectPattern = function (pattern) {
```
Comment (Snugug) [inline bin/sass-lint.js]: Codacy Issue found: [Function 'anonymous' has a complexity of 5.](https://www.codacy.com/app/snugug/sass-lint/file/2359663576/issues/source?bid=3169912&fileBranchId=3171901#l19)

**Comment by house9** [comment]
@DanPurdy merge conflicts have been resolved, Thanks.

**Comment by DanPurdy** [comment]
@house9 sorry for the delay here. This looks good to me, if you could update with the latest progress from develop then we can merge this in. Thanks! 👍

**Pull request merged**